src/AbstractDifferentiation.jl

```diff
@@ -480,8 +440,6 @@ macro primitive(expr)
         return define_jacobian_and_friends(fdef) |> esc
     elseif name == :primal_value
         return define_primal_value(fdef) |> esc
-    elseif name == :pullback_function
-        return define_pullback_function_and_friends(fdef) |> esc
```

Comment:
Why delete this?

Reply:
Because it's not supported anymore. It seems it was just forgotten to remove it in #93. I noticed it when fixing the merge conflicts. I could move it to a separate PR but I thought maybe it's fine to include this fix here.

Reply:
>  Why do we only have 2 out of 5 (now 4) ..._and_friends functions defined in the package? Where are the rest?

Good catch, the jacobian stuff is also missing. I'm not sure if the `jacobian_and_friends` has ever existed.

Reply:
So in the spirit of "simpler defaults", we might as well delete the rest?

Reply:
Ah, now I remember. The support for the jacobian macro was removed in https://github.com/JuliaDiff/AbstractDifferentiation.jl/pull/95. It seems it was reintroduced accidentally when fixing some merge conflicts?!

Reply:
So in the end, for a backend library to support AD, they either need to implement `pushforward_function` or `value_and_pullback_function`, from which the rest is deduced?
Making a mental note to document that and solve #91

Reply:
I guess you can use the macro and everything else will work. But I think usually that's not what you want. As already clear from the - probably incomplete - implementations for AD backends in this package and e.g. #97, good performance is only achieved if you implement all functions in the AbstractDifferentation interface for which the backend provides a - typically optimized - implementation.

The sole purpose of the macro is that you get an implementation of `jacobian` for free. But if the AD backend already defines a `jacobian` function, you can directly call it and don't even need that.

Reply:
Right, thanks! We can continue that conversation over on #91, in the meantime this is good to go